src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,29 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can`t be null!");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must contain at least 6 symbols!");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with this login already exists!");
+        }
+
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null!");
+        }
+
+        if (user.getPassword().length() < 6) {
+            throw new RegistrationException("Password must be contain at least 6 symbols!");
+        }
+
+        if (user.getAge() == null || user.getAge() < 18) {
```

Comment:
The check for null age is good, but make sure to handle it separately from the age value check to provide more specific exception messages.